### PR TITLE
KOGITO-5893: Data-index API connection with runtimes tasks: add comme…

### PR DESCRIPTION
**Pull request by nmirasch**
…nts and attachments

jira: https://issues.redhat.com/browse/KOGITO-5893

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by nmirasch** [comment]
Jenkins retest this

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java [hunk 1]
```diff
@@ -209,27 +231,36 @@ protected CompletableFuture sendDeleteClientRequest(WebClient webClient, String
...
+                .send(res -> asyncHttpResponseTreatment(res, future, logMessage));
+        return future;
+    }
+
+    protected CompletableFuture sendPostWithBodyClientRequest(WebClient webClient, String requestURI, String logMessage, String body, String contentType) {
+        CompletableFuture future = new CompletableFuture<>();
+        HttpRequest<Buffer> request = webClient.post(requestURI)
+                .putHeader("Authorization", getAuthHeader())
+                .putHeader("Content-Type", contentType);
+        if ("application/json".equals(contentType)) {
```
Comment (cristianonicolai) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java]: there must be a constant for this :)

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java [hunk 2]
```diff
@@ -123,8 +123,10 @@ private GraphQLSchema createSchema() {
                     builder.dataFetcher("NodeInstanceCancel", this::cancelNodeInstance);
                     builder.dataFetcher("JobCancel", this::cancelJob);
                     builder.dataFetcher("JobReschedule", this::rescheduleJob);
-                    builder.dataFetcher("TaskUpdate", this::updateUserTask);
-                    builder.dataFetcher("TaskPartialUpdate", this::partialUpdateUserTask);
+                    builder.dataFetcher("TaskInstanceUpdate", this::updateUserTask);
```
Comment (cristianonicolai) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java]:
```suggestion
                    builder.dataFetcher("UserTaskInstanceUpdate", this::updateUserTask);
```

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java (hunk 1 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java]: yep! I found that :)

**Comment by nmirasch** [comment]
@cristianonicolai renamed mutations and removed hard update. Replaced taskInfo parameter by the enumeration of available fields to update. Can you take a look when you have the chance? Thanks!

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java (hunk 2 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java]: done

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java [hunk 3]
```diff
@@ -176,24 +180,31 @@ protected WebClientOptions getWebClientToURLOptions(String targetHttpURL) {
...
-    public CompletableFuture<String> updateUserTask(String serviceURL, UserTaskInstance userTaskInstance, String user,
+    public CompletableFuture<String> updateUserTaskInstance(String serviceURL, UserTaskInstance userTaskInstance, String user,
             List<String> groups, Map taskInfo) {
-        String requestURI = format(UPDATE_TASK_PATH, userTaskInstance.getProcessId(), userTaskInstance.getProcessInstanceId(),
+        String requestURI = format(UPDATE_USER_TASK_INSTANCE_PATH, userTaskInstance.getProcessId(), userTaskInstance.getProcessInstanceId(),
                 userTaskInstance.getId()) + "?" + getUserGroupsURIParameter(user, groups);
-        return sendPutClientRequest(getWebClient(serviceURL),
-                requestURI,
-                "Update UserTask: " + userTaskInstance.getName() + " with id: " + userTaskInstance.getId(),
+        return sendPatchClientRequest(getWebClient(serviceURL), requestURI,
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java:187]: changing from put to patch ?

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java [hunk 4]
```diff
@@ -123,8 +123,9 @@ private GraphQLSchema createSchema() {
                     builder.dataFetcher("NodeInstanceCancel", this::cancelNodeInstance);
                     builder.dataFetcher("JobCancel", this::cancelJob);
                     builder.dataFetcher("JobReschedule", this::rescheduleJob);
-                    builder.dataFetcher("TaskUpdate", this::updateUserTask);
-                    builder.dataFetcher("TaskPartialUpdate", this::partialUpdateUserTask);
+                    builder.dataFetcher("UserTaskInstanceUpdate", this::updateUserTask);
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java]: this::updateUserTask -> this::updateUserTaskInstance ?

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java [hunk 5]
```diff
@@ -289,21 +290,30 @@ private CompletableFuture getUserTaskSchema(DataFetchingEnvironment env) {
...
     private CompletableFuture<String> updateUserTask(DataFetchingEnvironment env) {
         UserTaskInstance userTaskInstance = cacheService.getUserTaskInstancesCache().get(env.getArgument("taskId"));
-        Map taskData = env.getArgument("taskData");
-        return dataIndexApiExecutor.updateUserTask(getServiceUrl(userTaskInstance.getEndpoint(), userTaskInstance.getProcessId()),
+        return dataIndexApiExecutor.updateUserTaskInstance(getServiceUrl(userTaskInstance.getEndpoint(), userTaskInstance.getProcessId()),
                 userTaskInstance,
                 env.getArgument("user"),
                 env.getArgument("groups"),
-                env.getArgument("taskInfo"));
+                env.getArguments());
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:297]: all arguments from env ?

File: data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls [hunk 6]
```diff
@@ -356,17 +359,6 @@ input UserTaskInstanceMetaArgument {
     started: DateArgument
     referenceName: StringArgument
 }
-input UserTaskInfoArgument {
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls:359]: why removing that structure ?

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java (hunk 3 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java:187]: @radtriste not really. I've removed the old updateTask that uses the 'put' and only keep the 'partial update' that uses the patch. The difference is the put operation replaces all the fields related to the task and if they don't come in the variables the will be set to empty.. With this approach, only the parameters that came will be updated, it allows more control over what is affected by the update operation.

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java (hunk 5 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:297]: @radtriste  Here We have moved the task related arguments to update to upper level, removing the previous userTaskInfoArgument, letting the operation to pass just the fields that need to be updated. Here I'm passing all the arguments in a map and the operation itself will traduce that to updates if the parameter matches with any 'updatable' field. The schema is in charge to control that only those fields can be passed as arguments ( description, priority,..all the ones that before were contained in UserTaskInfoArgument )

**Review by radtriste — APPROVED** [review]
lgtm

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**